frontend/controller/actions/group.js

```diff
@@ -482,7 +482,16 @@ export default (sbp('sbp/selectors/register', {
       }
     } catch (e) {
       console.error('gi.actions/group/join failed!', e)
-      throw new GIErrorUIRuntimeError(L('Failed to join the group: {codeError}', { codeError: e.message }))
+      const primaryClicked = await sbp('gi.ui/prompt', {
+        heading: L('Failed to join the group'),
+        question: L('Error details:{br_}{err}', { err: e.message, ...LTags() }),
+        primaryButton: L('Refresh')
+      })
+
+      if (primaryClicked) {
+        const url = (window.location.href).split('?')[0]
+        window.open(url, '_self')
+      }
```

Comment:
In this PR, I think it's best to preserve the behavior of `master` as much as we can, as it will save both of us headaches on having to think through the consequences of changing the behavior.

Before there was no call to refresh the page, and no there shouldn't be either. It's good to change the behavior to have it show the error, but let's limit it to that for now unless there's a strong reason to do more.

Reply:
@taoeffect Will do so. I added the page refresh logic here as it was part of the suggested solution of the issue #2096, BTW.

Reply:
Hmm, but isn't #2096 about updating the PendingApproval page? 🤔

Reply:
@taoeffect 
Yes, I started off looking at the `PendingApproval.vue` but then realised `/pending-approval` page is merely the place redirected to as a follow-up step of an actual async action where the error is thrown, meaning `PendingApproval.vue` cannot catch that error.

For example,
`await sbp('gi.actions/group/joinWithInviteSecret')` [here](https://github.com/okTurtles/group-income/blob/41dc1f1d8ca0fea71f65a837d1ec717f522657e7/frontend/views/pages/Join.vue#L207C1-L209) leads to a chain of several `sbp` calls which involves `sbp('gi.actions/group/join')` ([here](https://github.com/okTurtles/group-income/blob/41dc1f1d8ca0fea71f65a837d1ec717f522657e7/frontend/model/contracts/identity.js#L257) in `identity.js` contract) down the line. that's actually where the target error mostly occurs. So I decided to add the prompt logic there instead.

So even after we land `/pending-approval` page, there is still a chain of async operations (which started before the page navigation) going on behind the scene. so the error thrown as part of that looks like it occurred within the pending approval page. This is what it appeared from my investigation.